Amir: Task 737-742: HEAD-QA, RO-STS-Parallel

shreeshiv opened the pull request:
@swarooprm Hi Swaroop, I have added six tasks. Can you review it?
Thanks.

swarooprm commented:
@amirrezamirzaei, wondering if you can help review this PR?

amirrezamirzaei commented:
Hi @shreeshiv , a few general comments
- remove the .DS_store file from the pr.
- change the tone of the definition in the ReadMe file to imperative. for example ` Language translation from English to Spanish (based on HEAD_QA)` could be `Given a sentence in English translate it to Spanish`.
- don't use the dataset name in the definitions and readMe file.
- the definition for the task is too long. you can omit the part explaining the dataset data. for example the definition for task 737 could be `In this task, You will be given a question about healthcare and a passage. You must choose the correct answer to the question from 4-5 different options on the basis of critical reasoning.` also it better if you use the format `In this task, You ...`.
- please use this order for the fields in each task -> (Contributors, Source, Categories, Definition, Positive Examples, Negative Examples, Instances)
- please add these fields to your translation tasks. ("Input_language" , "Output_language", "instruction_language") . see this [task](https://raw.githubusercontent.com/allenai/natural-instructions-expansion/master/tasks/task272_europarl_translation.json) for more info. please use the same order for these fields as in the task file. also, you can use the same definition too.

shreeshiv commented:
Thanks for your suggestions.
I will work on this and change is accordingly.

Best,
shreeshiv.

On Mon, Sep 20, 2021 at 11:14 AM amirrezamirzaei ***@***.***>
wrote:

> Hi @shreeshiv <https://github.com/shreeshiv> , a few general comments
>
>    - remove the .DS_store file from the pr.
>    - change the tone of the definition in the ReadMe file to imperative.
>    for example Language translation from English to Spanish (based on
>    HEAD_QA) could be Given a sentence in English translate it to Spanish.
>    - don't use the dataset name in the definitions and readMe file.
>    - the definition for the task is too long. you can omit the part
>    explaining the dataset data. for example the definition for task 737 could
>    be In this task, You will be given a question about healthcare and a
>    passage. You must choose the correct answer to the question from 4-5
>    different options on the basis of critical reasoning. also it better
>    if you use the format In this task, You ....
>    - please use this order for the fields in each task -> (Contributors,
>    Source, Categories, Definition, Positive Examples, Negative Examples,
>    Instances)
>    - please add these fields to your translation tasks. ("Input_language"
>    , "Output_language", "instruction_language") . see this task
>    <https://raw.githubusercontent.com/allenai/natural-instructions-expansion/master/tasks/task272_europarl_translation.json>
>    for more info. please use the same order for these fields as in the task
>    file. also, you can use the same definition too.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/swarooprm/CSE576HW/pull/93#issuecomment-923162741>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFMDZDGDD5MGSTCVRK2BFG3UC52XLANCNFSM5EBRZWWA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

swarooprm commented:
> Thanks for your suggestions. I will work on this and change is accordingly. Best, shreeshiv.
> […](#)
> On Mon, Sep 20, 2021 at 11:14 AM amirrezamirzaei ***@***.***> wrote: Hi @shreeshiv <https://github.com/shreeshiv> , a few general comments - remove the .DS_store file from the pr. - change the tone of the definition in the ReadMe file to imperative. for example Language translation from English to Spanish (based on HEAD_QA) could be Given a sentence in English translate it to Spanish. - don't use the dataset name in the definitions and readMe file. - the definition for the task is too long. you can omit the part explaining the dataset data. for example the definition for task 737 could be In this task, You will be given a question about healthcare and a passage. You must choose the correct answer to the question from 4-5 different options on the basis of critical reasoning. also it better if you use the format In this task, You .... - please use this order for the fields in each task -> (Contributors, Source, Categories, Definition, Positive Examples, Negative Examples, Instances) - please add these fields to your translation tasks. ("Input_language" , "Output_language", "instruction_language") . see this task <https://raw.githubusercontent.com/allenai/natural-instructions-expansion/master/tasks/task272_europarl_translation.json> for more info. please use the same order for these fields as in the task file. also, you can use the same definition too. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#93 (comment)](https://github.com/swarooprm/CSE576HW/pull/93#issuecomment-923162741)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFMDZDGDD5MGSTCVRK2BFG3UC52XLANCNFSM5EBRZWWA> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

@shreeshiv, can you address comment soon so that I can evaluate your HW?

shreeshiv commented:
Hi @swarooprm,
I have addressed suggestions. Can you please evaluate tasks? 
Thanking you.

Best,
shreeshiv.

swarooprm commented:
This tasks need to be revised E.g. "Since no contextual information about question, (i.e., passage) given in the dataset; model need to find answer from complex reasoning rather than specialized knowledge."--> Don't use terms like "model", "dataset" etc. Think how will you instruct your friend to do the task.

@amirrezamirzaei, can you check and let me know if you are happy or have any other feedback for @shreeshiv

shreeshiv commented:
> This tasks need to be revised E.g. "Since no contextual information about question, (i.e., passage) given in the dataset; model need to find answer from complex reasoning rather than specialized knowledge."--> Don't use terms like "model", "dataset" etc. Think how will you instruct your friend to do the task.
> 
> @amirrezamirzaei, can you check and let me know if you are happy or have any other feedback for @shreeshiv

Hi @swarooprm ,
I have revised task 737. Please check it. Let me know if further changes needed. 

Hi @amirrezamirzaei,Thanks for earlier suggestion, I revised it. Can you please suggest me if any changes needed?

Thanks and regards,
shreeshiv.

amirrezamirzaei commented:
few comments (they are mostly about the explanation of the examples)
- task 737:
1. the output of the instances needs to be the number of the choice not the entire text of the choice.
2. please remove the `Test name` and `category` from the input because they give no additional information.
3. the definition contains grammatical error please change it into something like `In this task your given a multi-choice question. Each question consists of five various choices. Your job is to find the correct choice using complex reasoning rather than specialized knowledge.`
4. the explanation for the examples contains a lot of grammatical errors for example in the first positive example explanation
the explanation should be `This question aims to find answers based on logical reasoning rather than contextual knowledge. The question is about tumor which is present in the second option ...`. please use online tools like Grammarly to evaluate these explanations.
5. don't mention the dataset in the third example explanation.
- task 738:
1. in the definition change `English sentence into the Spanish language` to `The English sentence into the Spanish language`
2. the explanations contain a lot of grammatical errors mostly you don't use "the". for example in the first positive example explanation ` Given input in English` should be `The given input in English` / `output must be negative` should be `the output must be negative` ...
- task 739:
1. in the definition change `Spanish sentence into the English language to `The Spanish sentence into the English language`
2. same comment as 738 concerning explanations

shreeshiv commented:
Hi @swarooprm,
I have revised task737 and modified others according to Amir's suggestions. Can you review it?

Thank and regards,
shreeshiv.

amirrezamirzaei commented:
thank you @shreeshiv there are still some grammatical errors:
- task 737:
1. please refer to the last comment concerning task 737 definition.
2. concerning first positive example change `The given question asked to find TUMOR which is present in the second option` to `the answer to the given question should be a tumor, therefore ...`
3. concerning second positive example change `Question asked` to `The question asked`. `answer would ..` to `the answer would ..`
4. concerning this task negative examples: the negative examples should have an incorrect output, not a bad input. For example, if the output should be the first choice instead the negative example output is the second choice.
- task 738
1. concerning first positive example:  `polarity of output` should be `polarity of the output`. `Looking at output it’s clear that sentence polarity is negative` should be `Looking at the output it’s clear that the sentence polarity is negative`.
2. concerning second positive example: there is no need to explain that we can use third-party application to evaluate. the explanation should be like the first example. if you cant think of anything just say `translation from English to Spanish looks good`.
3. concerning third positive example : why is there might in `language translation might be correct` ? 
4. concerning fourth and fifth positive example : `also plurality` should be `also the plurality`
5. concerning sixth positive example : same comment as in the third example
6. concerning first negative example: omit this ` Any quantitive measure, the polarity of sentences should remain same throughout the process`
7. change all instances of `looks good ` into `is correct`.
- task 739
1. concerning first positive example: omit `and looks good to me.` change `clear that sentence polarity` to `clear that the sentence polarity`
2. concerning second positive example: change `also plurality` to `also the plurality`.change `for a given task` to `for the given task`
3. concerning the third positive example: change `verb as the input` to `verb as the input`. same comment as before about `might be correct`. omit this part `The input contains the verb ”specialize” which is also present in output ”especializado”`
4. concerning the fourth positive example: change `also plurality` to `also the plurality`.
5. same comment as before concerning the two last positive examples.
6. concerning the first negative example: the explanation seem to be copied from the 738 and is incorrect
7. concerning the second negative example : change `should remain same` to `should remain the same`. change `Sentiment polarity both sentences are opposite of each other` to `the Sentiment polarity of both sentences are opposite of each other`
(breaking comments into another comment for ease of access)

shreeshiv commented:
Hi @amirrezamirzaei, Thanks for your suggestions. I appreciate your time.
I am working on tasks and push changes as soon as possible. Regarding the head_qa dataset, it is not a translation dataset but a QA dataset. But as it is available in multiple languages(Spanish and English), I'm using it for a translation task. In the dataset documentation, it is already mentioned that this translation is human-verified so I believe there won't be an issue in using it for the translation task.

amirrezamirzaei commented:
thank you for the changes. few minor comments:

- task 737: 
1. please refer to the comment about this task definition. (change definition to `In this task your given a multi-choice question. Each question consists of five various choices. Your job is to find the correct choice using complex reasoning rather than specialized knowledge.`).
2. the positive examples doesn't contain `### Options:` format. also please change  `### Options:` to ` Options:` or  `\nOptions:`
- task 738: remove `third party ..` from the sixth positive example. change`looks good` to `is correct` in the second positive example.
- task 739: change `polarity of output` to `polarity of the output` in the first positive example.  change `polarity both` to `polarity of both` in second negative example also change `quantitative measures, the polarity of sentences` to `quantitative measures, polarity of sentences` here.
- task 740: same comment about `### Options` as in 737. change `belongs to chemistry paper` to `belongs to the chemistry class` in second positive example. change `the Psychology exam paper.` to `the Psychology class.` in the third positive example.
- task 741: 
- change `output ` to `the output ` in second positive example. 
- change `it talks about the only plane` to `it talks about a plane` in first negative example.  also this isn't a very good negative example because we generally mean airplane when we use plane.
- change `input ` to `the input ` in second negative example.
- task 742:change `output ` to `the output ` in second positive example.same comment as 741 about the negative examples. change `1st ` to `first` in third negative example

shreeshiv commented:
Hi @amirrezamirzaei,
Regarding tasks 737 and 740, I used '###' as a delimiter between the question and options. I thought to put delimiter as ':' but some options already contain this character therefore I put '###'. If there is no need to put those special characters, let me know. I will revise it.
warm regards,
shreeshiv.

amirrezamirzaei commented:
please change the delimiter to '\n'.

- task 737 : looks good just change the delimiter
- task 738 : 
1. you missed a few comment. change ` looks good.` to ` is correct.`.
2. concerning first negative example:  change the word `convert ` to `translate`.`converted ` to `translated`. change `Thus, our translation changes quantitative measures while the translation process` to `Thus, our translation changes quantitative measures.`
3. ` the polarity of sentences should remain same` should be ` the polarity of sentences should remain the same`. please change everywhere(in other task too).
- task 739:
1. you missed a few comments. change ` output` to `the output` in the first positive example. change `maintain ` to `maintained` in second positve example.
- task 740: same comment concerning the delimeter
1. concerning second negative example change `The given question is taken from the nursery exam paper so ...` to `The given question is about nursery so ...`
- task 741&742: looks good! my only issue is the first negative example in both these tasks as I said in the previous comment. when we use plane we generally mean airplane so this isn't a very good negative example.

shreeshiv commented:
Hi @amirrezamirzaei  and @swarooprm ,
Thanks for the suggestions.
I did the following updates.
- Task 737, 740: Changed delimiter as you mentioned
- Task 738, 739, 740: Removed grammatical errors as you suggested.
- Task 741,742: Removed weak negative examples.

Thanks and regards.

amirrezamirzaei commented:
thank you @shreeshiv , everything looks good to me @swarooprm !
although I still have some concerns about  task 738&739 as pointed out [here](https://github.com/swarooprm/CSE576HW/pull/93#issuecomment-927901601). Im not very sure how we can evaluate the translation quality of these two tasks.

swarooprm commented:
Your HW2 is complete now.

Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @shreeshiv
